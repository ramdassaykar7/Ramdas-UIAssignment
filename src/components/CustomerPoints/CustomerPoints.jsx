import React from 'react';

/**
 * CustomerPoints Component
 * Displays rewards information for customers.
 * 
 * @param {Object} data - The rewards data structured by customer IDs.
 * @returns {JSX.Element} A component rendering customer rewards details or a fallback message.
 */
const CustomerPoints = ({ data }) => {
    // Check if data has any content
    const hasData = Object.keys(data).length > 0;

    return (
        <div>
            <h2>Customer Rewards</h2>
            {hasData ? (
                // Iterate over each customer ID in the data
                Object.entries(data).map(([customerId, { total, monthly }]) => (
                    <div key={customerId}>
                        <h3>Customer ID: {customerId}</h3>
                        <p>Total Points: {total}</p>
                        <ul>
                        {/* monthly points for each customer */}
                            {Object.entries(monthly).map(([month, points]) => (
                                <li key={month}>
                                    {month}: {points} points
                                </li>
                            ))}
                        </ul>
                    </div>
                ))
            ) : (
                //  When no data is available
                <p>No data available</p>
            )}
        </div>
    );
};

export default CustomerPoints;
