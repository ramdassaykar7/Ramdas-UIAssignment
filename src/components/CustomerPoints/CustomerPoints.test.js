import { render, screen } from '@testing-library/react';
import CustomerPoints from './CustomerPoints';
import '@testing-library/jest-dom';  // Ensure jest-dom matchers are available

const mockData = {
    1: { total: 150, monthly: { January: 90, February: 60 } },
    2: { total: 85, monthly: { January: 30, March: 55 } }
};

describe('CustomerPoints Component', () => {
    test('renders customer points correctly', () => {
        render(<CustomerPoints data={mockData} />);
        
        // Check for Customer 1 details
        expect(screen.getByText(/Customer ID: 1/i)).toBeInTheDocument();
        expect(screen.getByText(/Total Points: 150/i)).toBeInTheDocument();
        expect(screen.getByText(/January: 90 points/i)).toBeInTheDocument();
        expect(screen.getByText(/February: 60 points/i)).toBeInTheDocument();

        // Check for Customer 2 details
        expect(screen.getByText(/Customer ID: 2/i)).toBeInTheDocument();
        expect(screen.getByText(/Total Points: 85/i)).toBeInTheDocument();
        expect(screen.getByText(/March: 55 points/i)).toBeInTheDocument();
    });

    test('renders empty state when no data is provided', () => {
        render(<CustomerPoints data={{}} />);

        expect(screen.getByText(/No data available/i)).toBeInTheDocument();
    });

    test('renders correctly with partial data', () => {
        const partialData = {
            3: { total: 45, monthly: { March: 45 } }
        };
        render(<CustomerPoints data={partialData} />);

        expect(screen.getByText(/Customer ID: 3/i)).toBeInTheDocument();
        expect(screen.getByText(/Total Points: 45/i)).toBeInTheDocument();
        expect(screen.getByText(/March: 45 points/i)).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<CustomerPoints data={mockData} />);
        expect(asFragment()).toMatchSnapshot();
    });

  
});
