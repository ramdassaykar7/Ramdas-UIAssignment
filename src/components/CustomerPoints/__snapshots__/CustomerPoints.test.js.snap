// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerPoints Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Customer Rewards
    </h2>
    <div>
      <h3>
        Customer ID: 1
      </h3>
      <p>
        Total Points: 150
      </p>
      <ul>
        <li>
          January: 90 points
        </li>
        <li>
          February: 60 points
        </li>
      </ul>
    </div>
    <div>
      <h3>
        Customer ID: 2
      </h3>
      <p>
        Total Points: 85
      </p>
      <ul>
        <li>
          January: 30 points
        </li>
        <li>
          March: 55 points
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
